charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/statefulset.yaml

```
@@ -84,6 +78,11 @@ spec:
           httpGet:
             path: /_cluster/health
             port: http
+{{- if .Values.searchguard.enabled }}
+            httpHeaders:
+              - name: Authorization
+                value: Basic {{ .Values.elasticsearch.basicAuth }}
```

Alternatively, you could do the b64 part on chart side:
```suggestion
                value: Basic {{ .Values.elasticsearch.readinessProbe.httpAuth | b64enc }}
```

where elasticsearch.readinessProbe.httpAuth is in format user:pass